code/lib/builder-vite/src/index.ts

```diff
@@ -1,48 +1,31 @@
 // noinspection JSUnusedGlobalSymbols
 
 import * as fs from 'fs-extra';
-import type { Builder, StorybookConfig as StorybookBaseConfig, Options } from '@storybook/types';
 import type { RequestHandler } from 'express';
-import type { InlineConfig, UserConfig, ViteDevServer } from 'vite';
+import type { ViteDevServer } from 'vite';
 import express from 'express';
 import { dirname, join, parse } from 'path';
+import type { Options, StorybookConfig as StorybookBaseConfig } from '@storybook/types';
 import { transformIframeHtml } from './transform-iframe-html';
 import { createViteServer } from './vite-server';
 import { build as viteBuild } from './build';
-import type { ExtendedOptions } from './types';
+import type { ViteBuilder, StorybookConfigVite } from './types';
 
 export { withoutVitePlugins } from './utils/without-vite-plugins';
 export { hasVitePlugins } from './utils/has-vite-plugins';
 
-// TODO remove
-export type { TypescriptOptions } from '@storybook/types';
-
-// Storybook's Stats are optional Webpack related property
-export type ViteStats = {
-  toJson: () => any;
-};
-
-export type ViteBuilder = Builder<UserConfig, ViteStats>;
-
-export type ViteFinal = (
-  config: InlineConfig,
-  options: Options
-) => InlineConfig | Promise<InlineConfig>;
-
-export type StorybookConfig = StorybookBaseConfig & {
-  viteFinal?: ViteFinal;
-};
+export * from './types';
 
 /**
  * @deprecated
  *
- * Use `import { StorybookConfig } from '@storybook/builder-vite';`
+ * Import `StorybookConfig` from your framework, such as:
  *
- * Or better yet, import from your framework.
+ * `import type { StorybookConfig } from '@storybook/react-vite';`
  */
-export type StorybookViteConfig = StorybookConfig;
+export type StorybookViteConfig = StorybookBaseConfig & StorybookConfigVite;
```

Comment:
Keeping this for backwards-compat, but I'm also happy to remove it as a possibly-breaking change.  I did add a note in the `MIGRATION.md` about where the config should be imported from, but this could also stick around until 8.0 and be removed then.  I don't feel strongly either way.

Reply:
I think this can be removed.. I don't think it's worth keeping it around for another year / major release

Reply:
It looks like @tmeasday wants to keep it.  https://github.com/storybookjs/storybook/pull/20057

Reply:
ok 👍

Reply:
I think that request came from @shilman and .. ultimately vite folks?